src/models/hit-types/hit.ts

```diff
@@ -0,0 +1,54 @@
+import { ItemHit } from './item-hit';
+import { TextHit } from './text-hit';
+
+/**
+ * Union of the different hit_type values returned by the PPS.
+ * There will probably be more of these.
+ */
+export type HitType = 'item' | 'text';
+
+/**
+ * A type having all properties that exist on either T or U.
+ */
+type Merge<T, U> = {
+  [K in keyof T | keyof U]: K extends keyof T
+    ? T[K]
+    : K extends keyof U
+    ? U[K]
+    : never;
+};
+
+/**
+ * Applying this to the Hit type forces Intellisense to present Hit as
+ * a type in its own right, and not as the underlying merge type it aliases.
+ * Really just to keep things clean at the call site.
```

Comment:
curious as to why it does this when `export type Hit` ?

Reply:
I'm not certain myself actually (I would expect it to use the "nearest" type alias to the callsite). But I was seeing the messy-looking `Partial<Merge<blah>>` type getting exposed elsewhere so figured I'd override what Intellisense sees 🤷